cli/internal/runcache/runcache.go

```diff
@@ -164,17 +170,33 @@ func (tc TaskCache) RestoreOutputs(terminal *cli.PrefixedUi, logger hclog.Logger
 		}
 		return false, nil
 	}
-	// TODO(gsoltis): check if we need to restore goes here
-	// That will be an opportunity to prune down the set of outputs as well
-	hit, _, _, err := tc.rc.cache.Fetch(tc.rc.repoRoot.ToString(), tc.hash, tc.repoRelativeGlobs)
+	changedOutputGlobs, err := tc.rc.outputWatcher.GetChangedOutputs(tc.hash, tc.repoRelativeGlobs)
 	if err != nil {
-		return false, err
-	} else if !hit {
-		if tc.rc.cacheMissLogsMode != NoLogs {
-			terminal.Output(fmt.Sprintf("cache miss, executing %s", ui.Dim(tc.hash)))
+		logger.Warn(fmt.Sprintf("Failed to check if we can skip restoring outputs for %v: %v. Proceeding to check cache", tc.pt.TaskID, err))
+		terminal.Warn(ui.Dim(fmt.Sprintf("Failed to check if we can skip restoring outputs for %v: %v. Proceeding to check cache", tc.pt.TaskID, err)))
+		changedOutputGlobs = tc.repoRelativeGlobs
+	}
+	if len(changedOutputGlobs) == 0 {
```

Comment:
Can we name these "states" a bit more clearly? e.g. `hasChangedOutputs := len(changedOutputGlobs) != 0`

Reply:
Done, and flipped the conditional to `if hasChangedOutputs { ... } else { ... }`